pylint/checkers/imports.py

```diff
@@ -712,6 +717,24 @@ def _check_deprecated_module(self, node, mod_path):
             if mod_path == mod_name or mod_path.startswith(mod_name + '.'):
                 self.add_message('deprecated-module', node=node, args=mod_path)
 
+    def _check_import_as_rename(self, node):
+        names = node.names
+        if not all(names[0]):
+            return
+
+        real_name = names[0][0]
+        head, tail = os.path.splitext(real_name)
```

Comment:
Could we use rsplit here? We aren't splitting a file name so we don't need to use `splitext`. So this could be `head, tail = real_name.rsplit(1, '')`? Then you don't need to do `tail[1:]` either.

Reply:
I guess you meant, head, tail = real_name.rsplit('.', 1)
above will work fine for  head, tail = 'foo.bar.foobar'.rsplit('.', 1)
but fails for head, tail = 'OrderedDict'.rsplit('.', 1) case [from collections import OrderedDict as OrderedDict].
so will need to do atleast like below:
packages = real_name.rsplit('.', 1)
real_name = packages[1] if len(packages) == 2 else packages[0]
Anyways this is atleast more meaningful than using splitext.